frontend/src/pages/personal/vouchers/VotingRight.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import {
+  Box,
+  Typography,
+  Card,
+  CardContent,
+  Stack,
+  Tooltip,
+  IconButton,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+} from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
+import TaskAltIcon from '@mui/icons-material/TaskAlt';
+import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked';
+import { useTranslation, Trans } from 'react-i18next';
+
+import { AccountsService, UserProfile } from 'src/services/openapi';
+import { usePersonalVouchers } from './context';
+
+const Thumb = ({
+  up,
+  upDescription,
+  downDescription,
+}: {
+  up: boolean | undefined;
+  upDescription: string;
+  downDescription: string;
+}) => {
+  if (up === undefined) return null;
+
+  return (
+    <Tooltip title={up ? upDescription : downDescription}>
+      <IconButton>
+        {up ? (
+          <TaskAltIcon color="success" />
+        ) : (
+          <RadioButtonUncheckedIcon color="disabled" />
+        )}
+      </IconButton>
+    </Tooltip>
+  );
+};
+
+const DescriptionDialog = ({
+  open,
+  onClose,
+}: {
+  open: boolean;
+  onClose: () => void;
+}) => {
+  const { t } = useTranslation();
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <DialogTitle>{t('personalVouchers.votingRight.title')}</DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          <Trans t={t} i18nKey="personalVouchers.votingRight.description" />
+        </DialogContentText>
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+const VotingRight = () => {
+  const { t } = useTranslation();
+  const [userProfile, setUserProfile] = React.useState<
+    UserProfile | undefined
+  >();
+  const { receivedVouchers } = usePersonalVouchers();
+
+  React.useEffect(() => {
+    const loadUserProfile = async () => {
+      const userProfile = await AccountsService.accountsProfileRetrieve();
+      setUserProfile(userProfile);
+    };
+    loadUserProfile();
+  }, []);
+
+  const displayedValue = React.useMemo(() => {
+    if (userProfile === undefined) return undefined;
+    const { voting_right: votingRight } = userProfile;
```

Comment:
Done in d1d21cb1af72a99b304d17261219001af44acd57.

I did a merge instead of a rebase though (35c4c2b0b076b4aa63a50a95394d07c3dc3c4059). It seems much more natural to me. I don't see why I would rewrite the previous commit to make them look like they were written from the current main. Each of them would probably require some specific work to make them actually work on the new main, so I would most likely end up squashing them before rebasing, so they would be destroyed. By using a merge the history is consistent with what happened and there's no need to rewrite old commits.